cpp/src/copying/copy.cpp

```diff
@@ -69,6 +69,38 @@ gdf_column allocate_like(gdf_column const& input, bool allocate_mask_if_exists,
   return output;
 }
 
+/*
+ * Allocates a new column of the same size as the input with the specified type.
+ */
+gdf_column allocate_dtype(const gdf_column& input, gdf_dtype output_type, bool allocate_mask_if_exists, cudaStream_t stream)
```

Comment:
This function is a bit odd. It's just creating a new column of the same size as another column, but with a different type.

It'd be more general to create a function that creates a  `gdf_column` of a specified size and type.